app/helpers/thumbnail_helper.rb

```diff
@@ -59,6 +61,62 @@ def image_link_html(link = "", link_method = :get)
     end
   end
 
+  def image_caption_html(image_id, obs_data, link_type)
+    html = []
+    if obs_data[:id].present?
+      html = image_observation_data(html, obs_data, link_type)
+    end
+    html << caption_image_links(image_id)
+    safe_join(html)
+  end
+
+  def image_observation_data(html, obs_data, link_type)
+    if link_type == :naming || obs_data[:obs].vote_cache <= 0
+      html << caption_propose_naming_link(obs_data)
+    end
+    html << caption_obs_title(obs_data)
+    html << render(partial: "observations/show/observation",
+                   locals: { observation: obs_data[:obs] })
+    # render(partial: "observations/namings/form",
+    #        locals: { action: :create, url: url, show_reasons: true })
+  end
+
+  # Disabled because interpolation produces unsafe html
+  # rubocop:disable Style/StringConcatenation
+  def caption_image_links(image_id)
+    orig_url = Image.url(:original, image_id)
+    original_image_link(orig_url) + " | " + image_exif_link(image_id)
```

Comment:
I think the preference in such cases is safe_concat or something like that?  @JoeCohen, do you know the right syntax?  There's a way to do this that doesn't require disabling a cop.

Reply:
Fixed it with a safe_join